packages/xchain-ethereum/src/client.ts

```diff
@@ -601,16 +606,35 @@ export default class Client implements XChainClient, EthereumClient {
    *
    * @param {Address} spender The spender address.
    * @param {Address} sender The sender address.
+   * @param {feeOptionKey} FeeOptionKey Fee option (optional)
    * @param {BaseAmount} amount The amount of token. By default, it will be unlimited token allowance. (optional)
    * @returns {TransactionResponse} The transaction result.
    */
-  approve = async (spender: Address, sender: Address, amount?: BaseAmount): Promise<TransactionResponse> => {
+  approve = async (
+    spender: Address,
+    sender: Address,
+    feeOptionKey?: FeeOptionKey,
+    amount?: BaseAmount,
+  ): Promise<TransactionResponse> => {
     try {
+      const gasPrice =
+        feeOptionKey &&
+        BigNumber.from(
+          (
+            await this.estimateGasPrices()
+              .then((prices) => prices[feeOptionKey])
+              .catch(() => getDefaultGasPrices()[feeOptionKey])
+          )
+            .amount()
+            .toFixed(),
+        )
```

Comment:
I would avoid such nested things ^, since you can't know the result in case of errors. Even if you wrap everything into a `try/catch` statement before, the error message might be messed up and it might be hard to understand). Also, the code is not well readable anymore at some point. Just my 2 cents.